engine/src/main/java/org/camunda/bpm/engine/impl/cfg/CompositeProcessEnginePlugin.java

```diff
@@ -0,0 +1,131 @@
+
+/**
+ * {@link ProcessEnginePlugin} that provides composite behavior. When registered on an engine configuration,
+ * all plugins added to this composite will be triggered on preInit/postInit/postProcessEngineBuild.
+ * <p>
+ * Use to encapsulate common behavior (like engine configuration).
+ */
+public class CompositeProcessEnginePlugin extends AbstractProcessEnginePlugin {
+
+  protected final List<ProcessEnginePlugin> plugins;
+
+  /**
+   * New instance with empty list.
+   */
+  public CompositeProcessEnginePlugin() {
+    this.plugins = new ArrayList<ProcessEnginePlugin>();
+  }
+
+
+  /**
+   * New instance with vararg.
+   * @param plugin first plugin
+   * @param additionalPlugins additional vararg plugins
+   */
+  public CompositeProcessEnginePlugin(ProcessEnginePlugin plugin, ProcessEnginePlugin... additionalPlugins) {
+    this();
+    addProcessEnginePlugin(plugin, additionalPlugins);
+  }
+
+  /**
+   * New instance with initial plugins.
+   *
+   * @param plugins the initial plugins. Must not be null.
+   */
+  public CompositeProcessEnginePlugin(final List<ProcessEnginePlugin> plugins) {
+    this();
+    addProcessEnginePlugins(plugins);
+  }
+
+  /**
+   * Add one (or more) plugins.
+   *
+   * @param plugin first plugin
+   * @param additionalPlugins additional vararg plugins
+   * @return self for fluent usage
+   */
+  public CompositeProcessEnginePlugin addProcessEnginePlugin(ProcessEnginePlugin plugin, ProcessEnginePlugin... additionalPlugins) {
+    return this.addProcessEnginePlugins(toList(plugin, additionalPlugins));
+  }
+
+  /**
+   * Add collection of plugins.
+   *
+   * If collection is not sortable, order of plugin execution can not be guaranteed.
+   *
+   * @param plugins plugins to add
+   * @return self for fluent usage
+   */
+  public CompositeProcessEnginePlugin addProcessEnginePlugins(final Collection<ProcessEnginePlugin> plugins) {
+    this.plugins.addAll(plugins);
+
+    return this;
+  }
+
+  @Override
+  public void preInit(ProcessEngineConfigurationImpl processEngineConfiguration) {
+    for (ProcessEnginePlugin plugin : plugins) {
+      plugin.preInit(processEngineConfiguration);
+    }
+  }
+
+  @Override
+  public void postInit(ProcessEngineConfigurationImpl processEngineConfiguration) {
+    for (ProcessEnginePlugin plugin : plugins) {
+      plugin.postInit(processEngineConfiguration);
+    }
+  }
+
+  @Override
+  public void postProcessEngineBuild(ProcessEngine processEngine) {
+    for (ProcessEnginePlugin plugin : plugins) {
+      plugin.postProcessEngineBuild(processEngine);
+    }
+  }
+
+  /**
+   * Get all plugins.
+   *
+   * @return the configured plugins
+   */
+  public List<ProcessEnginePlugin> getPlugins() {
+    return plugins;
+  }
+
+  @Override
+  public String toString() {
+    return this.getClass().getSimpleName() + plugins;
+  }
+
+
+  private static List<ProcessEnginePlugin> toList(ProcessEnginePlugin plugin, ProcessEnginePlugin... additionalPlugins) {
```

Comment:
yes, you are right, this is only needed because I want to force usage of add(plugin) and disallow a parameterless add().